Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelWizard/ScanAPIModelWizardPage.xaml.cs

```diff
@@ -91,15 +122,77 @@ public void WizardEvent(WizardEventArgs WizardEventArgs)
                     Parse();
                 }
             }
+            else if (WizardEventArgs.EventType == EventType.LeavingForNextPage)
+            {
+                if (AddAPIModelWizard.DeltaModelsList != null && AddAPIModelWizard.DeltaModelsList.Count > 0)
+                {
+                    bool mergeIssue = false;
+                    bool notifyReplaceAPI = false;
+                    AddAPIModelWizard.LearnedAPIModelsList.Clear();
+                    foreach (DeltaAPIModel deltaModel in AddAPIModelWizard.DeltaModelsList.Where(m => m.IsSelected == true))
+                    {
+                        ApplicationAPIModel selectedAPIModel = null;
+
+                        if (deltaModel.DefaultOperationEnum == DeltaAPIModel.eHandlingOperations.Add)
+                            selectedAPIModel = deltaModel.learnedAPI;
+                        else if (deltaModel.DefaultOperationEnum == DeltaAPIModel.eHandlingOperations.MergeChanges)
+                        {
+                            if (deltaModel.MergedAPIModel == null)
+                            {
+                                mergeIssue = true;
+                                break;
+                            }
+                            else
+                            {
+                                deltaModel.MergedAPIModel.ContainingFolder = deltaModel.matchingAPIModel.ContainingFolderFullPath;
+                                deltaModel.MergedAPIModel.Guid = deltaModel.matchingAPIModel.Guid;
+
+                                selectedAPIModel = deltaModel.MergedAPIModel;
+                                notifyReplaceAPI = true;
+                            }
+                        }
+                        else if (deltaModel.DefaultOperationEnum == DeltaAPIModel.eHandlingOperations.ReplaceExisting)
+                        {
+                            deltaModel.learnedAPI.ContainingFolder = deltaModel.matchingAPIModel.ContainingFolderFullPath;
+                            deltaModel.learnedAPI.Guid = deltaModel.matchingAPIModel.Guid;
+
+                            selectedAPIModel = deltaModel.learnedAPI;
+                            notifyReplaceAPI = true;
+                        }
+
+                        if (selectedAPIModel != null)
+                            AddAPIModelWizard.LearnedAPIModelsList.Add(selectedAPIModel);
+                    }
+
+                    if(mergeIssue)
+                    {
+                        Reporter.ToUser(eUserMsgKey.StaticErrorMessage, "Merged API should be derived from a base API, Please update selected !");
```

Comment:
don't use staticErrormessage, you need to add message into ReporterMsgLib as Warning

Reply:
Done !